include/oneapi/dpl/pstl/glue_algorithm_ranges_impl.h

```diff
@@ -24,10 +24,228 @@
 #    include "hetero/algorithm_impl_hetero.h" //TODO: for __brick_copy
 #endif
 
+#if _ONEDPL_CPP20_RANGES_PRESENT
+#    include <ranges>
+#    include <functional>
+#    include "algorithm_ranges_impl.h"
+#endif
+
 namespace oneapi
 {
 namespace dpl
 {
+namespace ext
+{
+
+#if _ONEDPL_CPP20_RANGES_PRESENT
+namespace ranges
+{
+
+// [alg.foreach]
+
+struct for_each_fn
+{
+    template<typename _ExecutionPolicy, std::ranges::random_access_range _R, typename _Proj = std::identity,
```

Comment:
The biggest question I have for the concepts is whether we should have an (implementation defined) knob that allows users to turn them off. We have similar one in TBB and it seems pretty handy. It would require different spelling of concepts, though (not "require" strictly speaking but would be much preferred to not have a huge amount of macro

Reply:
Finally we decided don't follow "TBB way" and just added concept support.